scripts/format-changes.cmd

```diff
@@ -0,0 +1,22 @@
+@echo off
+setlocal
+setlocal EnableDelayedExpansion
+
+set PROJECT_ROOT=%~dp0\..
+
+set BRANCH=%1
+if "%BRANCH%"=="" (
+    echo ERROR: Missing commit/branch argument. If this is a fork of the microsoft/wil repo, you likely want to specify
+    echo        'upstream/master'. If this is not a fork, you likely want to specify 'origin/master'. Examples:
+    echo.
+    echo            format-changes.cmd origin/master

```

Comment:
On forks, I know that this is problematic if you've never run `git fetch upstream`, however the failure should hopefully be relatively obvious.

For both forks and non-forks this could potentially be problematic if you've run `git fetch upstream`/`git fetch origin`, but haven't merged with your local branch. Or similarly, if you've merged with master on a different machine and pulled only those changes without also fetching changes from master. I believe that doing so will end up diffing more than just the changes you've made in your branch. I don't believe that would necessarily be problematic (formatting more code shouldn't really affect much), but it might with mismatching `clang-format` versions. This should not be an issue for the CI check.

Reply:
Hopefully the readme is enough to diagnose any possible misuses